main/app/logger/log.cpp

```diff
@@ -0,0 +1,77 @@
+#include "log.hpp"
+#include <time.hpp>
+
+#define LOG_FILE_FORMAT "%d_log.txt"
+#define MAX_FILE_BYTE_SIZE (1024 * 1000)
+
+extern std::string string_vformat(const char* apFormat, ...);
+
+static constexpr const char logIds[4] = {
+    'D',
+    'I',
+    'W',
+    'E'
+};
+
+static const char* resetStyle = "\x1b[0m";
+
+static const char* ansiColors[4] = {
+    "\x1b[38;5;120m",
+    "\x1b[38;5;250m",
+    "\x1b[38;5;220m",
+    "\x1b[38;5;196m"
+};
+
+bool currentFile = false;
```

Comment:
wat doet dit precies?

Reply:
dit is voor het wisselen tussen twee bestanden voor het loggen, maar dit kan aangepast worden om iets te gebruiken als een timestamp voor de bestandsnamen (om ook logs te kunnen vergelijken op hoe oud ze zijn?)